### PR TITLE
Make Parquet reader filter APIs public (#1792)

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Part of #1792

# Rationale for this change
 

There is still more work to be done in this space, but I am confident that this functionality is sufficiently mature that we can expose it and start to get feedback from the wild

# What changes are included in this PR?

Makes the crate-private row filtering APIs, public

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/src/arrow/arrow_reader/selection.rs [hunk 1]
```diff
@@ -398,11 +398,11 @@ mod tests {
         let mut rand = thread_rng();
         for _ in 0..100 {
             let a_len = rand.gen_range(10..100);
-            let a_bools: Vec<_> = (0..a_len).map(|x| rand.gen_bool(0.2)).collect();
+            let a_bools: Vec<_> = (0..a_len).map(|_| rand.gen_bool(0.2)).collect();
```
Comment (alamb) [inline parquet/src/arrow/arrow_reader/selection.rs:401]: seems random that this started getting flagged by clippy 🤷  -- I suppose I am always one for a good drive by cleanup anyways 🚗

**Review by alamb — APPROVED** [review]
FYI @Ted-Jiang @thinkharderdev  @viirya and @HaoYang670 

This is quite exciting and represents a big step forward for the rust parquet implementation 🎉

File: parquet/src/arrow/arrow_reader/mod.rs [hunk 2]
```diff
@@ -141,14 +136,16 @@ impl<T> ArrowReaderBuilder<T> {
     }
 
     /// Provide a [`RowSelection] to filter out rows, and avoid fetching their
```
Comment (viirya) [inline parquet/src/arrow/arrow_reader/mod.rs:138]:
```suggestion
    /// Provide a [`RowSelection`] to filter out rows, and avoid fetching their
```

File: parquet/src/arrow/arrow_reader/selection.rs (hunk 1 above)
Comment (tustvold) [reply parquet/src/arrow/arrow_reader/selection.rs:401]: It is because the unused suppression at the module level was removed

**Pull request merged**